pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -301,13 +301,23 @@ func (a *ApplicationBackupController) handle(ctx context.Context, backup *stork_
 		}
 
 		// Try to create the backupLocation path, just log error if it fails
-		err := a.createBackupLocationPath(backup)
+		backupLocation, err := storkops.Instance().GetBackupLocation(backup.Spec.BackupLocation, backup.Namespace)
 		if err != nil {
-			log.ApplicationBackupLog(backup).Errorf(err.Error())
-			a.recorder.Event(backup,
-				v1.EventTypeWarning,
-				string(stork_api.ApplicationBackupStatusFailed),
-				err.Error())
+			return fmt.Errorf("error getting backup location path: %v", err)
+		}
+		if backupLocation.Location.Type != stork_api.BackupLocationNFS {
+			err := a.createBackupLocationPath(backup)
+			if err != nil {
+				log.ApplicationBackupLog(backup).Errorf(err.Error())
+				a.recorder.Event(backup,
+					v1.EventTypeWarning,
+					string(stork_api.ApplicationBackupStatusFailed),
+					err.Error())
+			}
+		}
+		// Below if condition just for debug purpose, Need to be removed.
+		if backupLocation.Location.Type == stork_api.BackupLocationNFS {
+			log.ApplicationBackupLog(backup).Infof("das: the nfs server address is %s and export path is %s", backupLocation.Location.NfsConfig.NfsServerAddr, backupLocation.Location.NfsConfig.NfsExportPath)
```

Comment:
Is "das:" relevant here?

Reply:
Oh it can be removed, I need to remove this line altogether in a final patch anyway..